dockerfiles/che/launcher.sh

```diff
@@ -0,0 +1,373 @@
+  VAL=$(if server_is_booted;then echo "YES"; else echo "NO"; fi)
+  debug "CHE SERVER IS BOOTED?     ${VAL}"
+  debug ""
+  debug "CHE_PORT                  = ${CHE_PORT}"
+  debug "CHE_VERSION               = ${CHE_VERSION}"
+  debug "CHE_RESTART_POLICY        = ${CHE_RESTART_POLICY}"
+  debug "CHE_USER                  = ${CHE_USER}"
+  debug "CHE_HOST_IP               = ${CHE_HOST_IP}"
+  debug "CHE_LOG_LEVEL             = ${CHE_LOG_LEVEL}"
+  debug "CHE_HOSTNAME              = ${CHE_HOSTNAME}"
+  debug "CHE_DATA_FOLDER           = ${CHE_DATA_FOLDER}"
+  debug "CHE_CONF_FOLDER           = ${CHE_CONF_FOLDER:-not set}"
+  debug "CHE_LOCAL_BINARY          = ${CHE_LOCAL_BINARY:-not set}"
+  debug ""
+  debug "---------------------------------------"
+  debug "---------------------------------------"
+  debug "---------------------------------------"
+}
+
+get_docker_install_type() {
+  if uname -r | grep -q 'boot2docker'; then
+    echo "boot2docker"
+  elif uname -r | grep -q 'moby'; then
+    echo "moby"
+  else
+    echo "native"
+  fi
+}
+
+get_docker_host_ip() {
+  NETWORK_IF="eth0"
+  INSTALL_TYPE=$(get_docker_install_type)
+  if [ "${INSTALL_TYPE}" = "boot2docker" ]; then
+    NETWORK_IF="eth1"
+  fi
+
+  docker run --rm --net host \
+            alpine sh -c \
+            "ip a show ${NETWORK_IF}" | \
+            grep 'inet ' | \
+            cut -d/ -f1 | \
+            awk '{ print $2}'
+}
+
+get_che_hostname() {
+  INSTALL_TYPE=$(get_docker_install_type)
+  CHE_IP=$(get_docker_host_ip)
+
+  if [ "${INSTALL_TYPE}" = "boot2docker" ]; then
+    echo "${CHE_IP}"
+  elif [[ "${INSTALL_TYPE}" = "moby" && "${CHE_IP}" = "10.0.75.2" ]]; then
+    echo "${CHE_IP}"
+  else
+    echo "localhost"
+  fi
+}
+
+check_docker() {
+  if [ ! -S /var/run/docker.sock ]; then
+    error_exit "Docker socket (/var/run/docker.sock) hasn't be mounted \
+inside the container. Verify the syntax of the \"docker run\" command."
+  fi
+
+  if ! docker ps > /dev/null 2>&1; then
+    output=$(docker ps)
+    error_exit "Error when running \"docker ps\": ${output}"
+  fi
+}
+
+che_container_exist() {
+  if [ "$(docker ps -aq  -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+che_container_is_running() {
+  if [ "$(docker ps -qa -f "status=running" -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+che_container_is_stopped() {
+  if [ "$(docker ps -qa -f "status=exited" -f "name=${CHE_SERVER_CONTAINER_NAME}" | wc -l)" = "0" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+wait_until_container_is_running() {
+  CONTAINER_START_TIMEOUT=${1}
+
+  ELAPSED=0
+  until che_container_is_running || [ ${ELAPSED} -eq "${CONTAINER_START_TIMEOUT}" ]; do
+    sleep 1
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+server_is_booted() {
+  if ! curl -v http://"${CHE_HOST_IP}":"${CHE_PORT}"/dashboard > /dev/null 2>&1 ; then
```

Comment:
`curl -v http://"${CHE_HOST_IP}":"${CHE_PORT}"/api` is ok? 

Should we fix https://github.com/eclipse/che/blob/master/Vagrantfile#L204 too?